### PR TITLE
Skipped Pagination for Locks index

### PR DESCRIPTION
**Pull request by sathishavm**
> Skip's pagination for the rendering Locks controller index method in JSON format.
> Limited to render only JSON format

### Risks
- Level: Low

### CODE REVIEW

File: app/controllers/resource_controller.rb [hunk 1]
```diff
@@ -115,7 +115,7 @@ def redirect_after_save
   end
 
   def search_resources
-    resource_class
+    resource_class.all
```
Comment (grosser) [inline app/controllers/resource_controller.rb]: why was this necessary ?

File: test/controllers/locks_controller_test.rb [hunk 2]
```diff
@@ -57,6 +57,21 @@ def create_lock(resource = nil, options = {})
...
 
+    describe '#index' do
+      it "doesn't paginate" do
+        2.times { create_lock stage }
+        get :index, format: :json, params: {per_page: 1}
+        json = JSON.parse(response.body)
+        json['locks'].count.must_equal 2
+      end
+
+      it "refuses to render html" do
```
Comment (grosser) [inline test/controllers/locks_controller_test.rb]: nice!

File: app/controllers/resource_controller.rb (hunk 1 above)
Comment (sathishavm) [reply app/controllers/resource_controller.rb]: Well this is optional for paginated resources, but for non-paginated resource, we need to list all the resources.
While during pagination this will get overridden by pagy.

**Comment by sathishavm** [comment]
> every page without a limit could one day be dead or be a DOS ... so prefer to keep pagination and make the client iterate

Agree, but the locks have API URL '/api/locks' which doesn't have good documentation about pagination for this URL and people started complaining about this.

File: app/controllers/resource_controller.rb (hunk 1 above)
Comment (grosser) [reply app/controllers/resource_controller.rb]: prefer to put that into the resoruce_controller so it's fixed in general

**Comment by grosser** [comment]
then improve the documentation ? :D
... make sure the reply also includes some meta attributes that says if this is the last page or so ...
